### PR TITLE
added temperature conversion in the doc equivalencies.rst

### PR DESCRIPTION
**Pull request by yashkmr99**
This pull request is in reference to Issue #8504 (Temperature Conversion)

### CODE REVIEW

File: docs/units/equivalencies.rst [hunk 1]
```diff
@@ -413,6 +413,24 @@ scaling:
     <Quantity 11.2254902 mag>
 
 
+Temperature Equivalency
+-----------------------
+
+The :func:`~astropy.units.temperature` equivalency allows conversion between the temperature units; degree Celsius, Fahrenheit and Kelvin.
+Example:
```
Comment (astrofrog) [inline docs/units/equivalencies.rst]: Maybe rephrase the end to say 'between the Celsius, Fahrenheit and Kelvin temperature units. For example::' (and make sure you put two colons (::) at the end so that the code block appears correctly.

**Comment by yashkmr99** [comment]
Okay done @astrofrog 
But I don't understand why circleci and travis-ci failed.

File: docs/units/equivalencies.rst (hunk 1 above)
Comment (yashkmr99) [reply docs/units/equivalencies.rst]: Ok done.
But I don't understand why travis-ci and circleci failed.

Comment (bsipocz) [reply docs/units/equivalencies.rst]: circle CI has failed with something unrelated, as you didn't start from the latest master.

the travis failure is due to the differences in float representations, try to add ``# doctest: +FLOAT_CMP`` to the end of L428

Comment (yashkmr99) [reply docs/units/equivalencies.rst]: Alright, I get it a bit. I should add just like this?
`<Quantity 32. deg_F>    # doctest: +FLOAT_CMP`

Comment (bsipocz) [reply docs/units/equivalencies.rst]: I messed up the line numbers. It should be on L427, the one that is executed rather than the output.

**Comment by bsipocz** [comment]
To get the circleCI fixed you need to rebase the PR. Some directions are here: http://docs.astropy.org/en/stable/development/workflow/development_workflow.html#rebase

Also check out the recent conversation on the #git channel.

File: docs/units/equivalencies.rst (hunk 1 above)
Comment (yashkmr99) [reply docs/units/equivalencies.rst]: Okay, I changed it. Thank you @bsipocz

File: docs/units/equivalencies.rst [hunk 2]
```diff
@@ -413,6 +413,23 @@ scaling:
     <Quantity 11.2254902 mag>
 
 
+Temperature Equivalency
+-----------------------
+
+The :func:`~astropy.units.temperature` equivalency allows conversion between the Celsius, Fahrenheit and Kelvin temperature units. For example::
```
Comment (pllim) [inline docs/units/equivalencies.rst]:
```suggestion
The :func:`~astropy.units.temperature` equivalency allows conversion
between Celsius, Fahrenheit, and Kelvin. For example::
```

File: docs/units/equivalencies.rst [hunk 3]
```diff
@@ -413,6 +413,23 @@ scaling:
...
+
+    >>> import astropy.units as u
+    >>> temp_C = 0*u.Celsius
```
Comment (pllim) [inline docs/units/equivalencies.rst]:
```suggestion
    >>> temp_C = 0 * u.Celsius
```

File: docs/units/equivalencies.rst [hunk 4]
```diff
@@ -413,6 +413,23 @@ scaling:
...
+    >>> temp_Kelvin = temp_C.to(u.K, equivalencies = u.temperature())
```
Comment (pllim) [inline docs/units/equivalencies.rst]:
```suggestion
    >>> temp_Kelvin = temp_C.to(u.K, equivalencies=u.temperature())
```

File: docs/units/equivalencies.rst [hunk 5]
```diff
@@ -413,6 +413,23 @@ scaling:
...
+    >>> temp_Kelvin
```
Comment (pllim) [inline docs/units/equivalencies.rst]:
```suggestion
    >>> temp_Kelvin  # doctest: +FLOAT_CMP
```

File: docs/units/equivalencies.rst [hunk 6]
```diff
@@ -413,6 +413,23 @@ scaling:
...
+    <Quantity 273.15 K>
+    >>> temp_F = temp_C.to(u.imperial.deg_F, equivalencies = u.temperature())
```
Comment (pllim) [inline docs/units/equivalencies.rst]:
```suggestion
    >>> temp_F = temp_C.to(u.imperial.deg_F, equivalencies=u.temperature())
```

File: docs/units/equivalencies.rst [hunk 7]
```diff
@@ -413,6 +413,23 @@ scaling:
...
+    >>> temp_F    # doctest: +FLOAT_CMP
```
Comment (pllim) [inline docs/units/equivalencies.rst]:
```suggestion
    >>> temp_F  # doctest: +FLOAT_CMP
```

File: docs/units/equivalencies.rst [hunk 8]
```diff
@@ -413,6 +413,23 @@ scaling:
...
+    <Quantity 32. deg_F>
+
+(Note : You can also use ``u.deg_C`` instead of ``u.Celsius``)
```
Comment (pllim) [inline docs/units/equivalencies.rst]:
```suggestion
.. note:: You can also use ``u.deg_C`` instead of ``u.Celsius``.
```

**Comment by sbrice** [comment]
@yashkmr99 what editor (i.e. PyCharm, vi, etc) are you using?

**Comment by yashkmr99** [comment]
Hi @sbrice, I use VSCode and gnu terminal but I am looking forward to move to something else, most probably vim.

**Comment by yashkmr99** [comment]
I have fixed the edits suggested @pllim. Sorry for the delay, I somehow skipped the mail regarding your requests.

**Comment by sbrice** [comment]
@bsipocz @pllim is there a recommended IDE setup? 

Something that would automatically break lines to the appropriate length, or ideally just an auto-formatter (e.g. https://medium.com/3yourmind/auto-formatters-for-python-8925065f9505)

**Comment by pllim** [comment]
@sbrice , we have general coding guidelines at http://docs.astropy.org/en/latest/development/codeguide.html and some Emacs specifics at http://docs.astropy.org/en/latest/development/codeguide_emacs.html .

I would **not** recommend setting your editor to *automatically* break lines, as sometimes it is necessarily to write longer than 80-char in a single line (e.g., inserting full URL). However, most editors have option to visualize where, say, 80-char limit is, if that helps. Emacs, Sublime, and other more popular editors also have option to do on-the-fly PEP 8 check with, say, `flake8`.

For further IDE discussions not related to this PR, let's chat over at Slack. http://joinslack.astropy.org/ :smile:
